test/duplicates.e2e-spec.ts

```diff
@@ -0,0 +1,933 @@
+import { WalletModule } from '../src/wallet';
+
+import { RepositoryModule } from '../src/contracts/repository';
+
+import { DepositService } from '../src/contracts/deposit';
+import { DepositModule } from '../src/contracts/deposit';
+
+import { SecurityModule, SecurityService } from '../src/contracts/security';
+
+import { LidoService } from '../src/contracts/lido';
+import { LidoModule } from '../src/contracts/lido';
+
+import { KeysApiService } from '../src/keys-api/keys-api.service';
+import { KeysApiModule } from '../src/keys-api/keys-api.module';
+
+import { ProviderService } from '../src/provider';
+import { GanacheProviderModule } from '../src/provider';
+
+import { BlsService } from '../src/bls';
+import { GuardianMessageService } from '../src/guardian/guardian-message';
+import { StakingModuleGuardService } from 'guardian/staking-module-guard';
+
+// Mock rabbit straight away
+jest.mock('../src/transport/stomp/stomp.client.ts');
+
+jest.setTimeout(10_000);
+
+describe('ganache e2e tests', () => {
+  let server: ReturnType<typeof makeServer>;
+
+  let providerService: ProviderService;
+  let walletService: WalletService;
+  let keysApiService: KeysApiService;
+  let guardianService: GuardianService;
+  let lidoService: LidoService;
+  let depositService: DepositService;
+  let blsService: BlsService;
+  let guardianMessageService: GuardianMessageService;
+  let securityService: SecurityService;
+  let sendDepositMessage: jest.SpyInstance;
+  let sendPauseMessage: jest.SpyInstance;
+  let stakingModuleGuardService: StakingModuleGuardService;
+  let getFrontRunAttempts: jest.SpyInstance;
+
+  beforeEach(async () => {
+    server = makeServer(FORK_BLOCK, CHAIN_ID, UNLOCKED_ACCOUNTS);
+    await server.listen(GANACHE_PORT);
+
+    if (!process.env.WALLET_PRIVATE_KEY) throw new Error(NO_PRIVKEY_MESSAGE);
+
+    const tempProvider = new ethers.providers.JsonRpcProvider(
+      `http://127.0.0.1:${GANACHE_PORT}`,
+    );
+
+    const wallet = new ethers.Wallet(
+      process.env.WALLET_PRIVATE_KEY,
+      tempProvider,
+    );
+
+    await wallet.sendTransaction({
+      to: SECURITY_MODULE_OWNER,
+      value: ethers.utils.parseEther('2'),
+    });
+  });
+
+  afterEach(async () => {
+    await server.close();
+  });
+
+  beforeEach(async () => {
+    // Prepare a signer for the unlocked Ganache account
+    if (!process.env.WALLET_PRIVATE_KEY) throw new Error(NO_PRIVKEY_MESSAGE);
+    const wallet = new ethers.Wallet(process.env.WALLET_PRIVATE_KEY);
+    const tempProvider = new ethers.providers.JsonRpcProvider(
+      `http://127.0.0.1:${GANACHE_PORT}`,
+    );
+    const tempSigner = tempProvider.getSigner(SECURITY_MODULE_OWNER);
+
+    // Add our address to guardians and set consensus to 1
+    const securityContract = SecurityAbi__factory.connect(
+      SECURITY_MODULE,
+      tempSigner,
+    );
+    await securityContract.functions.addGuardian(wallet.address, 1);
+
+    const moduleRef = await Test.createTestingModule({
```

Comment:
We have many lines of code in each test file. Let's move the logic of creating a test module to a separate file? This will significantly reduce the number of imports.